internal/topology/variables/cluster_variable_defaulting.go

```diff
@@ -86,83 +81,143 @@ func defaultClusterVariables(clusterVariables []clusterv1.ClusterVariable, clust
 		// Continue if there is no defaulted variable.
 		// NOTE: This happens when the variable doesn't exist on the CLuster before and
 		// there is no top-level default value.
-		if defaultedClusterVariable == nil {
+		if defaultedValue == nil {
 			continue
 		}
-
-		defaultedClusterVariables = append(defaultedClusterVariables, *defaultedClusterVariable)
+		defaultedValues = append(defaultedValues, *defaultedValue)
 	}
 
 	if len(allErrs) > 0 {
 		return nil, allErrs
 	}
+	return defaultedValues, nil
+}
 
-	return defaultedClusterVariables, nil
+// getCurrentValue returns the value of a variable for its definitionFrom, or for an empty definitionFrom if it exists.
+func getCurrentValue(variable clusterv1.ClusterVariable, valuesMap map[string]map[string]clusterv1.ClusterVariable) *clusterv1.ClusterVariable {
+	// If the value is set in the Cluster spec get the value.
+	if valuesForName, ok := valuesMap[variable.Name]; ok {
+		if value, ok := valuesForName[variable.DefinitionFrom]; ok {
+			return &value
+		}
+		if value, ok := valuesForName[emptyDefinitionFrom]; ok {
+			return &value
+		}
```

Comment:
Wondering if we need / should have that fallback. Could it be that DefinitionFrom is already either set to an empty or non-empty value and thus we don't need the fallback here?

Reply:
I think I realized later that we need it. But please double check

Reply:
I think you're right with the changes in getAllVariables.